tests/integrations/litestar/test_litestar.py

```diff
@@ -0,0 +1,326 @@
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = [
+        {
+            "op": "middleware.litestar",
+            "description": "SampleMiddleware",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+    ]
+    for idx, span in enumerate(transaction_event["spans"]):
+        assert span["op"] == expected[idx]["op"]
+        assert span["description"] == expected[idx]["description"]
+        assert span["tags"] == expected[idx]["tags"]
+
+
+def test_middleware_receive_send(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SampleReceiveSendMiddleware])
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        # NOTE: the assert statements checking
+        # for correct behaviour are in `SampleReceiveSendMiddleware`!
+        client.get("/message")
+    except Exception:
+        pass
+
+
+def test_middleware_partial_receive_send(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SamplePartialReceiveSendMiddleware])
+    events = capture_events()
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        client.get("/message")
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = [
+        {
+            "op": "middleware.litestar",
+            "description": "SamplePartialReceiveSendMiddleware",
+            "tags": {"litestar.middleware_name": "SamplePartialReceiveSendMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.receive",
+            "description": "TestClientTransport.create_receive.<locals>.receive",
+            "tags": {"litestar.middleware_name": "SamplePartialReceiveSendMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SamplePartialReceiveSendMiddleware"},
+        },
+    ]
+
+    for idx, span in enumerate(transaction_event["spans"]):
+        assert span["op"] == expected[idx]["op"]
+        assert span["description"].startswith(expected[idx]["description"])
+        assert span["tags"] == expected[idx]["tags"]
```

Comment:
Same comment, let's avoid indexing manually and let's make the test order-independent unless order matters